R/translate-sql-clause.r

```diff
@@ -40,6 +40,25 @@ sql_clause_limit <- function(limit, con){
   }
 }
 
+sql_clause_sample <- function(sample, con){
+  if (!is.null(sample$size) && !identical(sample$size, Inf)) {
+    assert_that(is.numeric(sample$size), length(sample$size) == 1L, sample$size >= 0)
+    if(sample$type == "n"){
+      build_sql(
```

Comment:
I had to use `format()` to ensure that small fractions are not converted to scientific notation:

```
size <- 0.000000000001
sql_expr(TABLESAMPLE (!! size), con = simulate_dbi())
<SQL> TABLESAMPLE(1e-12)
```

I used `trunc()` to prevent passing `10.0` instead of `10`. 

Would you recommend a different way?

Reply:
This is my current solution, it works for both cases:

```
      size <- format(sample$size, scientific = FALSE)
      sql_expr(TABLESAMPLE (!! sql(size)), con = con)
```